### DISCUSSION TITLE
Request to add Cloudflare Zero Trust Tunnel

### DISCUSSION THREAD

**Discussion by webmasterkai** [SmoothWAN/SmoothWAN#96] [Ideas and feature requests]
Similar to Tailscale, a [Cloudflare tunnel](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/install-and-setup/tunnel-guide/remote/#set-up-a-tunnel-remotely-dashboard-setup) allows public access to services running behind NAT.
It's a more convenient option when domains/project are already DNS/hosted with Cloudflare. I'm imagining the option that allows remote configuration using the Cloudflare Zero Trust Tunnel dashboard.

It requires the [cloudflared service/package](https://github.com/cloudflare/cloudflared/releases) to be installed. Once a tunnel is configured in the Cloudflare dashboard it requires running the `cloudflared service install [base64 encoded json string]`. Perhaps a webform in the SmoothWAN interface could allow the user to paste in the base64 string.

**Comment by TalalMash** [comment]
Fortunately it's built in OpenWRT mainline, LuCI integration should be simple, added next release.

**Reply by TalalMash** [reply]
Pretty tricky to get it working with online dashboard/ZTC side of things for the average user (exclusion, route setup, proxy etc), perhaps when it's in that level of 'advanced' a CLI approach wouldn't be inconvenient, tested this great guide and it worked well: https://hmolina.dev/p/cloudflared-tunnel-in-openwrt/ 
Note that this is a proxy service, UDP works poorly and no ICMP which may affect few software (remote desktop etc)

**Reply by webmasterkai** [reply]
Yeah, a CLI approach is totally fine. That's a nice guide you found.
Thank you!
